File: problems/2027-minimum-moves-to-convert-string/solution.py
class Solution:
    def minimumMoves(self, s: str) -> int:
        n = len(s)
        moves = 0
        i = 0
        while i < n:
            if s[i] == 'X':
                moves += 1
                i += 3
            else:
                i += 1
        return moves

File: problems/2027-minimum-moves-to-convert-string/solution.java
class Solution {
    public int minimumMoves(String s) {
        int n = s.length();
        int moves = 0;
        for (int i = 0; i < n; ) {
            if (s.charAt(i) == 'X') {
                moves++;
                i += 3;
            } else {
                i++;
            }
        }
        return moves;
    }
}